aspnetcore/fundamentals/logging-2.0.md

```diff
@@ -0,0 +1,126 @@
+
+```csharp
+services.AddLogging(builder => builder
+                .AddConsole()
+                .AddDebug();
+```
+
+In addition to now being part of the DI system, we have added methods on `WebHostBuilder` in your `Program.cs` to allow you to configure logging there instead of in your `Startup.cs`:
+
+```csharp
+var builder = new WebHostBuilder()
+                .UseKestrel()
+                .UseContentRoot(Directory.GetCurrentDirectory())
+                .ConfigureAppConfiguration((hostingContext, config) =>
+                {
+                    var env = hostingContext.HostingEnvironment;
+                    config.AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                          .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
+                    config.AddEnvironmentVariables();
+                })
+                .ConfigureLogging((hostingContext, logging) =>
+                {
+                    logging.AddConfiguration(hostingContext.Configuration.GetSection("Logging"));
+                    logging.AddConsole();
+                    logging.AddDebug();
+                });
+```
+
+A big advantage shown in this example is using the `HostingContext` provided to the `ConfigureLogging` extension method. This allows easy access to the configuration that is also registered in `Program.cs`. This leaves your `Startup.cs` more focused on configuring services and your middleware pipeline with less other concerns (namely logging and configuration).
+
+The final thing that impacts configuring logging is the new opinionated `WebHost` that you can see in the ASP.NET Core 2.0 templates. The default `WebHost` will configure logging in the same way as the sample above, meaning that if you only need `Console` and `Debug` logging and will use a `Logging` configuration section to control things like MinimumLogLevel or filters, then you don't need to add any extra code. If you want to add additional `ILoggerProviders` then you can call `ConfigureLogging` as well, adding just the providers that you need:
+
+```csharp
+WebHost.CreateDefaultBuilder(args)
+    .UseStartup<Startup>()
+    .ConfigureLoggin(logging => logging.AddEventLog())
+    .Build();
+```
+In the above sample you have all of the default loggers (Console and Debug) as well as the EventLog provider that you are adding.
+
+## Configuring Filtering
+
+In ASP.NET Core 1.x there are ways to do filtering, but they are either not very intuitive or are limited to specific log providers. In 2.0 we have changed the default `ILoggerFactory` to support filtering, meaning that you can filter log messages to all providers, and allow binding the filtering to `IConfiguration` allowing you to control it from whatever configuration sources you choose, such as a file.
```

Comment:
Sentence is too long:
"to support filtering. This means that..." 
"and allows binding the filtering to `IConfigurationg`."
"This new approach lets you control filtering from whatever configuration sources you choose (for example, a settings file)."